src/common/snippets/src/op/serialization_node.cpp

```diff
@@ -40,22 +40,48 @@ std::shared_ptr<Node> SerializationNode::clone_with_new_inputs(const OutputVecto
 }
 
 bool SerializationNode::visit_attributes(AttributeVisitor &visitor) {
+    auto is_planar_layout = [](const std::vector<size_t>& layout) {
+        for (size_t i = 0; i < layout.size(); ++i)
+            if (layout[i] != i) return false;
+        return true;
+    };
+
     std::vector<size_t> in_regs, out_regs;
     std::vector<std::string> in_reg_types, out_reg_types;
-    std::vector<std::pair<std::string, std::vector<size_t>>> shapes;
+    std::vector<std::pair<std::string, VectorDims>> shapes;
+    std::vector<std::pair<std::string, VectorDims>> subtensors;
+    std::vector<std::pair<std::string, std::vector<size_t>>> layouts;
     for (size_t i = 0; i < m_expr->get_input_count(); i++) {
         const auto& desc = m_expr->get_input_port_descriptor(i);
-        const auto &shape = desc->get_shape();
+        const auto& shape = desc->get_shape();
```

Comment:
Maybe it's better to convert this shape to `ov::PartialShape` before serialization? In this case, we will get more readable info in case of dynamic shapes: instead of `in_shape_0={18446744073709551615, 18446744073709551615}` we will get `in_shape_0={?,?}`. What do you think?

Reply:
Good idea, converted to `PartialShape` in https://github.com/openvinotoolkit/openvino/pull/24289/commits/cbfb2c7532dd37742b4cb5e042a52c82a52e91aa. Thanks!